src/gc.c

```diff
@@ -3599,6 +3611,8 @@ JL_DLLEXPORT void *jl_gc_counted_malloc(size_t sz)
             jl_atomic_load_relaxed(&ptls->gc_num.allocd) + sz);
         jl_atomic_store_relaxed(&ptls->gc_num.malloc,
             jl_atomic_load_relaxed(&ptls->gc_num.malloc) + 1);
+        jl_atomic_fetch_add_relaxed(&gc_heap_stats.bytes_mallocd, sz);
+        jl_atomic_fetch_add_relaxed(&gc_heap_stats.heap_size, sz);
```

Comment:
We might be able to measure this affect on Nanosoldier (for BigInt tests), but also might want to test it ourselves. It seems like at least one of these counter updates shouldn't be necessary, since we already do have the ptls data for it.

Reply:
Do we have a multithreaded bigint test?

Reply:
The bigint test we have in GC benchmarks is singlethreaded and it doesn't show much difference, I'll make it threaded and see if there's a difference.

Reply:
So on an admitedly very synthetic benchmark it does show up as a quite big regression